Assets/MRTK/Providers/LeapMotion/Editor/LeapMotionConfigurationChecker.cs

```diff
@@ -403,21 +415,87 @@ static void UpdateCSC()
+                    "https://developer.leapmotion.com/unity"
+                     , "OK");
+            }
+
+            ConfigureLeapMotion(isLeapInProject);
+        }        
+        
+        /// <summary>
+        /// Disconnect MRTK and the Leap Motion Unity Modules and display a prompt for the user to close unity and delete the assets.
+        /// </summary>
+        [MenuItem("Mixed Reality Toolkit/Utilities/Leap Motion/Disconnect Leap Motion Unity Modules")]
+        public static void DisconnectLeapMotion()
+        {
+            // If the user tries to disconnect the Leap Assets without assets in the project display a message
+            if (!isLeapInProject)
+            {
+                EditorUtility.DisplayDialog("MRTK Leap Motion Removal", "There are no Leap Motion Unity Modules in the project to disconnect, please add the Leap Motion Unity " +
+                    " Modules to use the MRTK's Leap Motion Data Provider", "OK");
+
+                return;
+            }
+
+            isLeapInProject = false;
+
+            // Force removal of the ScriptingDefinitions while the Leap Assets are still in the project
+            ScriptUtilities.RemoveScriptingDefinitions(BuildTargetGroup.Standalone, Definitions);
+            ScriptUtilities.RemoveScriptingDefinitions(BuildTargetGroup.WSA, Definitions);
 
+            // Remove the references to the Leap assembly definitions within the Leap Provider assembly definition 
             ConfigureLeapMotion(isLeapInProject);
+
+            // Prompt the user to close unity and delete the assets to completely remove.  Closing unity and deleting the assets is optional.
+            EditorUtility.DisplayDialog("MRTK Leap Motion Removal", "To complete the removal of the Leap Motion Unity Modules, close Unity, delete the assets, and reopen Unity", "OK");
+        }
+
+        /// <summary>
+        /// Uses the references in the Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef to determine if the integration of 
+        /// the Leap Motion Unity Modules was successful.
+        /// </summary>
+        /// <returns>True if the Leap Motion Unity Modules are integrated, false otherwise</returns>
+        private static bool GetConnectionStatus()
+        {
+            FileInfo[] leapDataProviderAsmDefFile = FileUtilities.FindFilesInAssets("Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef");
+
+            AssemblyDefinition leapDataProviderAsmDef = AssemblyDefinition.Load(leapDataProviderAsmDefFile[0].FullName);
+
+            bool isLeapConnected = leapDataProviderAsmDef.References.Contains("LeapMotion");
+
+            return isLeapConnected;
+        }
+
+        /// <summary>
+        /// Check the integration status of the Leap Motion Assets and display a message to the user.
+        /// </summary>
+        [MenuItem("Mixed Reality Toolkit/Utilities/Leap Motion/Check Connection Status")]
```

Comment:
Is the word "connection" overloaded in any way (i.e. as in being connected to the actual device, vs being hooked up w/ the SDK itself?)

I don't know if there's a better word though - but do you know what I mean here?

Reply:
Yes I do know what you mean, maybe we could consider using integration instead of connection?

Reply:
Looks good!